### PR TITLE
Porting sequence softmax operator

### LINKED ISSUE #4471: Porting Sequence Softmax Operator

### PR DESCRIPTION
**Pull request by Xreki**
resolve #4471 
It is the basic implementation of sequence softmax operator, which calling `Eigen` to do all the computations. I am not sure how many times CUDA kernel will be launched in `Eigen`. I will try figure it out and optimize the implementation in next PR.

### CODE REVIEW

File: paddle/operators/sequence_softmax_op.cc [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X",
+             "(LoDTensor) 1-D or 2-D input LoDTensor with the 2-nd dimension "
+             "of length 1.");
+    AddOutput("Out",
+              "(LoDTensor) 1-D or 2-D output LoDTensor with the 2-nd dimension "
+              "of length 1.");
+    AddComment(R"DOC(
+SequenceSoftmaxOp computes softmax activation among all time-steps for each
+sequences. The dimension of each time-step should be 1. Thus, the shape of
```
Comment (lcy-seso) [inline paddle/operators/sequence_softmax_op.cc]: for each sequence.

File: paddle/operators/sequence_softmax_op.cc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+input Tensor can be either [N, 1] or [N], where N is the sum of all sequences'
+length.
```
Comment (lcy-seso) [inline paddle/operators/sequence_softmax_op.cc]: sum of all sequences' lengths.

File: paddle/operators/sequence_softmax_op.cc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Equation:
+    for i-th sequence in mini-batch:
```
Comment (lcy-seso) [inline paddle/operators/sequence_softmax_op.cc]: in a mini-batch

**Review by lcy-seso — COMMENTED** [review]
LGTM, and please merge the latest dev branch.

File: paddle/operators/sequence_softmax_op.cc (hunk 1 above)
Comment (Xreki) [reply paddle/operators/sequence_softmax_op.cc]: Done.

File: paddle/operators/sequence_softmax_op.cc (hunk 2 above)
Comment (Xreki) [reply paddle/operators/sequence_softmax_op.cc]: Done.

File: paddle/operators/sequence_softmax_op.cc (hunk 3 above)
Comment (Xreki) [reply paddle/operators/sequence_softmax_op.cc]: Done.

**Review by lcy-seso — APPROVED** [review]
LGTM.

**Pull request merged**